Modules/vc-module-licensing/VirtoCommerce.LicensingModule.Data/Services/LicenseService.cs

```diff
@@ -0,0 +1,200 @@
+                        if (targetEntity != null)
+                        {
+                            changedEntries.Add(new GenericChangedEntry<License>(entity, originalLicense, EntryState.Modified));
+                            sourceEntity.Patch(targetEntity);
+                        }
+                        else
+                        {
+                            repository.Add(sourceEntity);
+                            changedEntries.Add(new GenericChangedEntry<License>(entity, EntryState.Added));
+                        }
+                    }
+                }
+
+                
+                await repository.UnitOfWork.CommitAsync();
+                await _eventPublisher.Publish(new LicenseChangedEvent(changedEntries));
+                pkMap.ResolvePrimaryKeys();
+            }
+        }
+
+        public async Task DeleteAsync(string[] ids)
+        {
+            using (var repository = _licenseRepositoryFactory())
+            {
+                await repository.RemoveByIdsAsync(ids);
+                await repository.UnitOfWork.CommitAsync();
+            }
+        }
+
+        public async Task<string> GetSignedLicenseAsync(string code, string clientIpAddress, bool isActivated)
+        {
+            string result = null;
+
+            var licenseEntity = await GetByCodeAsync(code);
+            if (licenseEntity != null)
+            {
+                var license = new
+                {
+                    licenseEntity.Type,
+                    licenseEntity.CustomerName,
+                    licenseEntity.CustomerEmail,
+                    licenseEntity.ExpirationDate,
+                };
+
+                var licenseString = JsonConvert.SerializeObject(license);
+                var signature = CreateSignature(licenseString);
+
+                result = string.Join("\r\n", licenseString, signature);
+
+                //Raise event
+                var activateEvent = new LicenseSignedEvent(licenseEntity.ToModel(AbstractTypeFactory<License>.TryCreateInstance()), clientIpAddress, isActivated);
+                await _eventPublisher.Publish(activateEvent);
+            }
+
+            return result;
+        }
+
+
+        private async Task<LicenseEntity> GetByCodeAsync(string code)
+        {
+            using (var repository = _licenseRepositoryFactory())
+            {
+                return await repository.Licenses.FirstOrDefaultAsync(x => x.ActivationCode == code);
+            }
+        }
+
+        private string CreateSignature(string data)
+        {
+            var hashAlgorithmName = HashAlgorithmName.SHA256.Name;
+
+            using (var hashAlgorithm = SHA256.Create())
+            using (var rsa = new RSACryptoServiceProvider())
+            {
+                // TODO: Store private key in a more secure storage, for example in Azure Key Vault
+                var privateKey = _settingsManager.GetValue("Licensing.SignaturePrivateKey", string.Empty);
```

Comment:
Could we avoid using the `settingmanager` for store secret key? Can we use the same solutions as we use for store and obtain public key in the platform?